packages/bot/src/stores/main-content-store.js

```diff
@@ -0,0 +1,42 @@
+import {
+    action,
+    observable,
+    reaction,
+}                                  from 'mobx';
+import { tabs_title }              from '../constants/bot-contents';
+import { onWorkspaceResize }       from '../scratch/utils/workspace';
+
+export default class MainContentStore {
+    constructor(root_store) {
+        this.root_store = root_store;
+        const { run_panel } = this.root_store;
+
+        reaction(
```

Comment:
- This `reaction` should be disposed when app unloads.
- Can we move the `if` condition in `setChartContainerSize` to the reaction definition?